packages/button/src/LoadingButton.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import BlockUi from 'react-block-ui';
+import { Button, ButtonProps } from 'reactstrap';
+
+type Props = {
+  id?: string;
+  isLoading: boolean;
+  children: string;
+  className?: React.ReactChild;
+  type?: string;
+} & ButtonProps;
+
+const LoadingButton = ({ id, isLoading, className, children, type, ...rest }: Props): JSX.Element => (
+  <Button
+    data-testid="loading_button"
+    id={id}
+    disabled={isLoading}
+    type={type}
+    className={`button ${isLoading ? 'is-loading' : ''} ${className || ''}`}
+    {...rest}
+  >
+    <BlockUi tag="div" blocking={isLoading} keepInView />
+    {!isLoading && <>{children}</>}
+  </Button>
+);
+
+LoadingButton.defaultProps = {
```

Comment:
We already have a lot of components using this style of `Component.defaultProps`, but it has been deprecated and we should start moving away from it in favor of default assignments

https://github.com/facebook/react/pull/16210
https://github.com/yannickcr/eslint-plugin-react/issues/2396#issuecomment-539184761

Reply:
Good to know!